cmd/thanos/store.go

```diff
@@ -279,11 +279,15 @@ func runStore(
 	}
 
 	// Ensure we close up everything properly.
-	defer func() {
-		if err != nil {
-			runutil.CloseWithLogOnErr(logger, bkt, "bucket client")
-		}
-	}()
+	{
```

Comment:
I'm trying to follow the code - it looks like we're also closing `bkt` in another run group below at [365](https://github.com/thanos-io/thanos/blob/main/cmd/thanos/store.go#L365). Do we need both?

Also, are other components using bucket clients adhering to this or are they using simply `defer` instead as well?

Reply:
You are correct, I have missed that one. We were talking about this particular code snippet that I've edited in our meetings with Prem & Akansha so I haven't even bothered to check other places. With this, we can simply remove this code block. I don't know about all clients but I think it's customary to call Close() at the end of using some resource.